ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java

```diff
@@ -779,10 +786,42 @@ private static TopologyAndConfigs readTopologyFile(final String file, final Obje
     }
   }
 
-  static Stream<JsonTestCase> findTestCases(final Path dir) {
+  private static List<Path> getTests(final Path dir, final List<String> files) {
+    List<Path> filePaths = new ArrayList<>();
+    for (String file : files) {
+      filePaths.add(dir.resolve(file.trim()));
+    }
+
+    return filePaths;
+  }
+
+  /**
+   * Returns a list of files specified in the system property 'ksql.test.file'.
+   * The list may be specified as a comma-separated string.
+   */
+  public static List<String> getTestFilesParam() {
+    String propValue = System.getProperty(KSQL_TEST_FILES);
+    if (propValue == null || propValue.trim().isEmpty()) {
+      return null;
+    }
+
+    return Arrays.asList(propValue.trim().split(","));
+  }
+
+  static Stream<JsonTestCase> findTestCases(final Path dir, final List<String> files) {
     final ClassLoader classLoader = EndToEndEngineTestUtil.class.getClassLoader();
 
-    return findTests(dir).stream()
+    List<Path> testPaths;
```

Comment:
nit: i'd probably put this into a method `getTestPaths(..)`
and then do `return getTestPaths(files).stream()...` or similar

Reply:
Done. It made the code more clear.